File: LeetCode/78子集.go
var ans [][]int
var path []int

func subsets(nums []int) [][]int {
	path = []int{}
	ans = [][]int{}
	// sort.Ints(nums)
	backtracking(nums, 0)
	return ans
}

func backtracking(nums []int, ind int) {
	temp := make([]int, len(path))
	copy(temp, path)
	ans = append(ans, temp)
	if ind >= len(nums) {
		return
	}
	for i := ind; i < len(nums); i++ {
		path = append(path, nums[i])
		backtracking(nums, i+1)
		path = path[:len(path)-1]
	}
}

File: LeetCode/78子集.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    LinkedList<Integer> path = new LinkedList<>();
    public List<List<Integer>> subsets(int[] nums) {
        backtracking(nums, 0);
        return ans;
    }

    public void backtracking(int[] nums, int ind) {
        ans.add(new ArrayList<>(path));
        if (ind > nums.length) return ;
        for (int i = ind; i < nums.length; i++) {
            path.add(nums[i]);
            backtracking(nums, i + 1);
            path.removeLast();
        }
        return ;
    }
}